src/vs/workbench/contrib/terminalContrib/accessibility/browser/terminalAccessibleBuffer.ts

```diff
@@ -260,26 +258,75 @@ export class AccessibleBufferWidget extends DisposableStore {
 		this._editorWidget.focus();
 	}
 
-	private _getContent(startIndex?: number): string {
+	private async _updateModel(): Promise<ITextModel> {
+		if (this._lastMarker?.isDisposed) {
+			this._lines = [];
+		}
+		this._removeViewportContent();
+		this._updateScrollbackContent();
+		this._updateViewportContent();
+		let model = this._editorWidget.getModel();
+		const text = this._lines.join('\n');
+		if (model) {
+			this._logService.debug('Edited accessible buffer model with ', this._lines.length, ' lines');
+			model.setValue(text);
+		} else {
+			this._logService.debug('Created new accessible buffer model with ', this._lines.length, ' lines');
+			model = await this._getTextModel(this._instance.resource.with({ fragment: text }));
+		}
+		this._editorWidget.setModel(model);
+		this._lastMarker = this._xterm.raw.registerMarker();
+		this._lastRowCount = this._xterm.raw.rows;
+		return model!;
+	}
+
+	private _removeViewportContent(): void {
+		if (this._lines.length && this._lastMarker?.line) {
+			// remove previous viewport content in case it has changed
+			for (let i = this._lastMarker.line; i < this._lastMarker.line + this._lastRowCount; i++) {
+				this._lines.pop();
+			}
```

Comment:
This might cause loss of data as `_lastMarker.line` and `_lastRowCount` is the wrapped line index, but `_lines` is unwrapped?

Reply:
hm, yeah good point. maybe I need to increment `i` by the number of newlines within the line that is removed?

Reply:
never mind, there aren't new lines in there so need some kind of mapping system

Reply:
this was a confusing way to write the code, but `_lines` is the wrapped lines and we want to `pop` the last number of lines that comprised the viewport